### PR TITLE
Add debug cookies feature – Part 2

### PR DESCRIPTION
**Pull request by renanferrari**
This PR adds a "Debug cookies" option to the Debug Settings screen. It basically allows us to add and remove any cookie value to be used on our app requests for testing purposes:

https://user-images.githubusercontent.com/830056/128933846-cdd386ae-d049-41f7-b5d0-3133e392a06c.mp4

### To test

Notes: 
- You'll need a tool like Charles so you can verify that the cookies are being added to the requests.

1. Open the app.
1. Tap the Me button.
1. On the Me screen, go to App Settings > Debug settings > Debug cookies.
1. On the Debug Cookies screen, add any kind of cookie name and cookie value combination (ex: `my_cookie_name` and `my_cookie_value`).
1. Tap "Set cookie".
1. Go back to My Site screen.
1. On Charles, verify that new API requests have the new cookie added to them:

## Regression Notes
1. Potential unintended areas of impact
None that I can think of.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing and existing unit tests.

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/debug/cookies/DebugCookiesAdapter.kt [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.DiffUtil.Callback
+import androidx.recyclerview.widget.RecyclerView.Adapter
+import androidx.recyclerview.widget.RecyclerView.ViewHolder
+import org.wordpress.android.databinding.DebugCookieItemBinding
+import org.wordpress.android.ui.debug.cookies.DebugCookiesAdapter.DebugCookieItemViewHolder
+import org.wordpress.android.ui.utils.ListItemInteraction
+import org.wordpress.android.util.viewBinding
+
+class DebugCookiesAdapter : Adapter<DebugCookieItemViewHolder>() {
```
Comment (ravishanker) [inline WordPress/src/main/java/org/wordpress/android/ui/debug/cookies/DebugCookiesAdapter.kt]: Same here as part 1, List Adapter in place of Adapter if you like

File: WordPress/src/main/java/org/wordpress/android/ui/debug/cookies/DebugCookiesFragment.kt [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+import android.widget.EditText
+import androidx.appcompat.app.AppCompatActivity
+import androidx.lifecycle.ViewModelProvider
+import androidx.recyclerview.widget.LinearLayoutManager
+import dagger.android.support.DaggerFragment
+import org.wordpress.android.R
+import org.wordpress.android.databinding.DebugCookiesFragmentBinding
+import javax.inject.Inject
+
+class DebugCookiesFragment : DaggerFragment(R.layout.debug_cookies_fragment) {
```
Comment (ravishanker) [inline WordPress/src/main/java/org/wordpress/android/ui/debug/cookies/DebugCookiesFragment.kt]: DaggerFragment use might make it hard when upgrading to Fragment or another DI,  unless it is really required to be used it here

File: WordPress/src/main/res/layout/debug_cookies_activity.xml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.fragment.app.FragmentContainerView xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (ravishanker) [inline WordPress/src/main/res/layout/debug_cookies_activity.xml:2]: Nice!! FragmentContainerView

File: WordPress/src/main/java/org/wordpress/android/ui/debug/cookies/DebugCookiesAdapter.kt (hunk 1 above)
Comment (renanferrari) [reply WordPress/src/main/java/org/wordpress/android/ui/debug/cookies/DebugCookiesAdapter.kt]: Updated in 08f7b390956ab1e060f159cae9cd3b88b204fa8c.

File: WordPress/src/main/java/org/wordpress/android/ui/debug/cookies/DebugCookiesFragment.kt (hunk 2 above)
Comment (renanferrari) [reply WordPress/src/main/java/org/wordpress/android/ui/debug/cookies/DebugCookiesFragment.kt]: Good point! Updated in f851ab819427ff710f83f325c9f2c3b9639ffe04.

**Comment by renanferrari** [comment]
Note: we should update the FluxC tag before merging this.

**Review by ravishanker — APPROVED** [review]

**Pull request merged**